### PR TITLE
Cisco Secure Access - add formats

### PR DESCRIPTION
**Pull request by lvoloshyn-sekoia**
https://github.com/SekoiaLab/detection-rules/issues/3917

## Type of Change

- [X] New intake format
- [ ] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product:

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Add new Cisco Secure Access intake module with parsers and metadata for web, cloud firewall, DNS, IPS, and file event logs.

New Features:
- Introduce ECS-aligned web proxy log parser and custom fields for Cisco Secure Access Web.
- Add cloud firewall log parser, custom fields, and manifest for Cisco Secure Access Cloud Firewall events.
- Add DNS log parser, custom fields, and manifest for Cisco Secure Access DNS resolver traffic.
- Add IPS log parser, custom fields, and manifest for Cisco Secure Access intrusion prevention events.
- Add file events log parser, custom fields, and manifest for Cisco Secure Access file inspection events.

Documentation:
- Add manifests and skeleton changelogs for Cisco Secure Access parent module and all new sub-formats.

Tests:
- Add initial JSON-based parser test fixtures for all new Cisco Secure Access formats.

### CODE REVIEW

File: CiscoSecureAccess/cisco-secure-access-cloud-firewall/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          cisco_secure_access.firewall.destination_list_ids: "{{parsed_dsv.message.destination_list_ids.split(',')}}"
+          cisco_secure_access.firewall.fw_event_id: "{{parsed_dsv.message.fw_event_id}}"
+          cisco_secure_access.firewall.app_id: "{{parsed_dsv.message.app_id}}"
+          cisco_secure_access.firewall.private_app_group_id: "{{parsed_dsv.message.private_app_group_id}}"
+          cisco_secure_access.firewall.private_flow: "{{parsed_dsv.message.private_flow}}"
+          cisco_secure_access.firewall.posture_id: "{{parsed_dsv.message.posture_id}}"
+          cisco_secure_access.firewall.casi_category_ids: "{{parsed_dsv.message.casi_category_ids}}"
+          cisco_secure_access.firewall.traffic_source: "{{parsed_dsv.message.traffic_source}}"
+          cisco_secure_access.firewall.content_category_ids: "{{parsed_dsv.message.content_category_ids.split(',')}}"
+          cisco_secure_access.firewall.content_category_list_ids: "{{parsed_dsv.message.content_category_list_ids.split(',')}}"
```
[the remark this answers is not in the document]
Comment (mchupeau-sk) [reply CiscoSecureAccess/cisco-secure-access-cloud-firewall/ingest/parser.yml:129]: @lvoloshyn-sekoia can you check this remark ?

File: CiscoSecureAccess/cisco-secure-access-file-events/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+name: cisco-secure-access-file-actions
```
[the remark this answers is not in the document]
Comment (mchupeau-sk) [reply CiscoSecureAccess/cisco-secure-access-file-events/ingest/parser.yml]: @lvoloshyn-sekoia can you check this remark ?

File: CiscoSecureAccess/_meta/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+uuid: ba321fe8-ec1f-4019-9a55-764b54dd9bae
+name: Cisco Secure Access
+slug: "cisco-secure-access"
+description: >-
+  Cisco Secure Access is a cloud-delivered security service edge solution combining secure web gateway, cloud-delivered firewall, DNS-layer security, cloud access security broker, zero trust network access and intrusion prevention into a single offering.
```
[the remark this answers is not in the document]
Comment (mchupeau-sk) [reply CiscoSecureAccess/_meta/manifest.yml:5]: @lvoloshyn-sekoia can you check this remark ?

File: CiscoSecureAccess/cisco-secure-access-web/tests/test_web.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
+{
+  "input": {
+    "message": "\"2025-01-06 13:39:24\",\"SIG2-Denver\",\"1.1.1.1\",\"4.4.4.4\",\"4.4.4.4\",\"\",\"ALLOWED\",\"https://example.com\",\"\",\"\",\"200\",\"1557\",\"6441\",\"\",\"\",\"Software/Technology,Allow List,Computers and Internet\",\"\",\"\",\"\",\"\",\"\",\"Network Tunnels\",\"\",\"SIG2-EX,RTO-XXXX-SIG\",\"Network Tunnels,Internal Networks\",\"CONNECT\",\"\",\"\",\"\",\"13368132\",\"610352\",\"9999583\",\"not_isolated\",\"\",\"\",\"\",\"\",\"\",\"\",\"\",\"nginx-proxy-abc123.example.net\",\"DEN1\",\"true\",\"winatp-gw-eus.microsoft.com\",\"false\",\"false\",\"\",\"\",\"REDACTED\",\"\",\"\",\"3.3.3.3\",\"deepseek-v3\",\"Prohibited Suppliers, Copyleft License\",\"e846afe0f3c8a87f\""
```
[the remark this answers is not in the document]
Comment (mchupeau-sk) [reply CiscoSecureAccess/cisco-secure-access-web/tests/test_web.json:3]: @lvoloshyn-sekoia can you check this remark ?

**Comment by mchupeau-sk** [comment]
@lvoloshyn-sekoia it's an automated review so check if it's revelant or not !

## ❌ Critical Issues (must fix)

### 1. Parser name mismatch in `cisco-secure-access-file-events`

**File:** `CiscoSecureAccess/cisco-secure-access-file-events/ingest/parser.yml`, line 1

The `name:` field says `cisco-secure-access-file-actions` but the format slug (folder name) is `cisco-secure-access-file-events`. The name must match the slug.

```yaml
# Current (wrong)
name: cisco-secure-access-file-actions

# Fix
name: cisco-secure-access-file-events
```

### 2. Missing vendor-level `README.md`

**File:** `CiscoSecureAccess/README.md` — does not exist.

The repository requires a `README.md` at the vendor directory level. All other vendor directories (e.g. `Cisco/README.md`, `Fortinet/README.md`) have one. It should describe the vendor and list the available formats.

### 3. Missing `observer.type` in DNS, Web, and File-events parsers

`observer.type` is a required ECS field for all parsers. It is correctly set in cloud-firewall (`firewall`) and IPS (`ids`) but is absent from three parsers:

| Format | Suggested `observer.type` value |
|---|---|
| `cisco-secure-access-dns` | `"dns"` |
| `cisco-secure-access-web` | `"proxy"` |
| `cisco-secure-access-file-events` | `"proxy"` |

```yaml
# Fix — add to the initial set block of each parser
observer.type: "proxy"   # or "dns"
```

---

## ⚠️ Warnings (non-blocking)

### 4. Field type `number` not accepted by taxonomy validator

**File:** `CiscoSecureAccess/cisco-secure-access-file-events/_meta/fields.yml`

Four fields declare `type: number`. The accepted types are: `keyword`, `long`, `float`, `boolean`, `date`, `array`, `object`, `ip`, `match_only_text`. Use `long` for integer counts and `float` for decimal scores:

| Field | Current type | Suggested type |
|---|---|---|
| `cisco_secure_access.file.archive_depth` | `number` | `long` |
| `cisco_secure_access.file.retention_policy_days` | `number` | `long` |
| `cisco_secure_access.file.threat_score` | `number` | `long` |
| `cisco_secure_access.file.type_id` | `number` | `long` |

### 5. `cisco_secure_access.firewall.start_time` typed as `keyword` instead of `date`

**File:** `CiscoSecureAccess/cisco-secure-access-cloud-firewall/_meta/fields.yml`

The parser writes an ISO 8601 timestamp to `cisco_secure_access.firewall.start_time` via `| to_rfc3339`. The `type` should be `date`, not `keyword`:

```yaml
cisco_secure_access.firewall.start_time:
  description: ...
  name: cisco_secure_access.firewall.start_time
  type: date   # was: keyword
```

### 6. `cisco_secure_access.firewall.packet_size` stores a byte count but is typed `keyword`

**File:** `CiscoSecureAccess/cisco-secure-access-cloud-firewall/_meta/fields.yml`

The description says "size in bytes". Numeric fields should use `long` for proper aggregations:

```yaml
cisco_secure_access.firewall.packet_size:
  type: long   # was: keyword
```

### 7. Large `set:` blocks without null guards for string fields

Several parsers set many ECS fields in a single `set:` block without `filter:` guards. Because `dsv.parse-dsv` returns empty strings for empty CSV columns (not `None`), fields like `organization.id`, `rule.id`, `cloud.region`, `destination.domain`, `network.direction`, `event.action`, etc. will be populated as empty strings when the source column is empty.

**Cloud-firewall example** (`set_ecs_fields` stage, second set block):
```yaml
# These fields have no null/empty guard:
destination.domain: "{{parsed_dsv.message.fqdns.split(',') | first}}"
cloud.region: "{{parsed_dsv.message.aws_region}}"
organization.id: "{{parsed_dsv.message.organization_id}}"
rule.id: "{{parsed_dsv.message.rule_id}}"
event.action: "{{parsed_dsv.message.action | lower}}"
```

Recommended pattern:
```yaml
- set:
    cloud.region: "{{parsed_dsv.message.aws_region}}"
  filter: "{{parsed_dsv.message.aws_region | length > 0}}"
```

This affects all 5 parsers to varying degrees.

### 8. `destination.domain` split may produce empty string

**File:** `CiscoSecureAccess/cisco-secure-access-cloud-firewall/ingest/parser.yml`

```yaml
destination.domain: "{{parsed_dsv.message.fqdns.split(',') | first}}"
```

If the `fqdns` CSV field is empty, this evaluates to `""`. Add a filter:

```yaml
- set:
    destination.domain: "{{parsed_dsv.message.fqdns.split(',') | first}}"
  filter: "{{parsed_dsv.message.fqdns | length > 0}}"
```

### 9. `event.start` not set in cloud-firewall parser

The parser captures `first_packet_timestamp` only in the custom field `cisco_secure_access.firewall.start_time`. The standard ECS field `event.start` should also be populated:

```yaml
event.start: "{{parsed_dsv.message.first_packet_timestamp | to_rfc3339}}"
```

### 10. Missing `relationships` in all `smart-descriptions.json` files

None of the 5 smart-descriptions files include a `relationships` array. The skill requires each entry to have `value`, `conditions`, and `relationships`. Example for cloud-firewall:

```json
{
  "value": "{event.action} {network.direction} traffic from {source.ip}:{source.port} to {destination.ip}:{destination.port}",
  "conditions": [
    { "field": "event.action" },
    { "field": "network.direction" },
    { "field": "source.ip" },
    { "field": "source.port" },
    { "field": "destination.ip" },
    { "field": "destination.port" }
  ],
  "relationships": [
    { "source": "source.ip", "target": "destination.ip" }
  ]
}
```

---

## 🚨 Security Issues

### IP addresses in test files not from RFC 5737 TEST-NET ranges

**Affected files:** All test files in all 5 formats.

All test files use the IPs `1.1.1.1`, `2.2.2.2`, `3.3.3.3`, `4.4.4.4`. The policy requires using only TEST-NET ranges for IPs in test data on this public repository:
- **Required ranges:** `192.0.2.x`, `198.51.100.x`, `203.0.113.x`

`1.1.1.1` is the real Cloudflare public DNS resolver. While the other IPs (`2.2.2.2`, `3.3.3.3`, `4.4.4.4`) are commonly used as placeholder test IPs, they do not comply with the RFC 5737 / repository policy.

**Recommended replacements:**

| Current | Suggested |
|---|---|
| `1.1.1.1` (client/source) | `192.0.2.10` |
| `2.2.2.2` (NAT/egress) | `192.0.2.20` |
| `3.3.3.3` (server/destination) | `198.51.100.10` |
| `4.4.4.4` | `203.0.113.10` |

All test `expected` objects and `related.ip` arrays would need to be updated accordingly.

---

**Review by mchupeau-sk — APPROVED** [review]
LGFM
